trace/HTTP.md

```diff
@@ -117,3 +117,35 @@ known attributes/labels on supported tracing backends.
 | "http.route"              | "http.route"       | "http.route"       | "/http/route"             |
 | "http.user_agent"         | "http.user_agent"  | "http.user_agent"  | "/http/user_agent"        |
 | "http.status_code"        | "http.status_code" | "http.status_code" | "/http/status_code"       |
+
+## Sampling
+
+There are two ways to control the `Sampler` used:
+* Controlling the global default `Sampler` via [TraceConfig](https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/TraceConfig.md).
+* Pass a specific `Sampler` as an option to the HTTP plugin. Plugins should support setting
```

Comment:
Can you please elaborate on scenarios when per-request sampler is needed?

Reply:
One scenario may be a sampler that is registered by exporter to filter out calls to that exporter endpoint. Is it intended use for this API?

Reply:
BTW, can this callback be used to extract additional information from http object and set a new attribute on span?

Reply:
The simplest scenario is to always sample when `?debug=true` or however that is propagated through a system. Other scenarios might be to always turn on tracing for developer requests or specific clients.

Reply:
I added some examples, PTAL.

The only use case is not just disabling tracing but being able to give user availability to use different policies per path. E.g. some endpoints with low traffic may get a higher sampling rate.

This is why I think it is better to influence the sampling policy here rather than building exporter filters. We also care about the sampling option to be set because it will be propagated.

@derekperkins, thanks for the example. I added it to the list as well.

Reply:
@SergeyKanzhelev I proposed this https://github.com/census-instrumentation/opencensus-specs/pull/186 as a way to pass attributes from the Sampler to the Span.